weather_providers/meteireann.py

```diff
@@ -0,0 +1,196 @@
+                                24: "Light rain showers and thunder",
+                                25: "Heavy rain showers and thunder",
+                                26: "Light sleet showers and thunder",
+                                27: "Heavy sleet showers and thunder",
+                                28: "Light snow showers and thunder",
+                                29: "Heavy snow showers and thunder",
+                                30: "Light rain and thunder",
+                                31: "Light sleet and thunder",
+                                32: "Heavy sleet and thunder",
+                                33: "Light snow and thunder",
+                                34: "Heavy snow and thunder",
+                                40: "Drizzle",
+                                41: "Light rain showers",
+                                42: "Light sleet showers",
+                                43: "Heavy sleet showers",
+                                44: "Light snow showers",
+                                45: "Heavy snow showers",
+                                46: "Drizzle",
+                                47: "Light sleet",
+                                48: "Heavy sleet",
+                                49: "Light snow",
+                                50: "Heavy snow"
+
+                            }
+        description = description_dict[weathercode]
+
+        logging.debug(
+            "get_description_by_weathercode({}) - {}"
+            .format(weathercode, description))
+
+        return description.title()
+
+
+    def hour_offset_from_now(self, h):
+        selected_hour = datetime.datetime.utcnow() + timedelta(hours=h)
+        return selected_hour.strftime("%Y-%m-%dT%H:00:00Z")
+
+
+    # Get weather from Met Eireann's open data API:
+    # https://data.gov.ie/dataset/met-eireann-weather-forecast-api
+    def get_weather(self):
+        url = ("http://metwdb-openaccess.ichec.ie/metno-wdb2ts/locationforecast?lat={};long={}"
+               .format(self.location_lat, self.location_long))
+
+        root = self.get_response_xml(url)
+
+        # the document contains the next day's forecast, in 24 hour-offset datapoints (as
+        # well as longer term forecasts, but we only want the next day).
+	# scan across the next 24 hours looking for high and low temperatures
+        temps = []
+        for h in range(0, 23):
+            hour_string = self.hour_offset_from_now(h)
+            for t in root.findall("./product/time[@from='%s']/location/temperature" % (hour_string)):
+                temps.append(float(t.get("value")))
+
+        temps.sort()
+        temperatureMin = temps[0]
+        temperatureMax = temps[-1]
+
+        # unfortunately, there's no daily summary field in the document; instead,
+	# get the symbol for just under 1 hour from now (so it's a very near-term
+        # forecast!).  TODO: maybe more than 1 hour would be better?
```

Comment:
possibly this could be configurable.... thoughts?

Reply:
I think leave it for now, unless there's some need for configuring it, or some need to choose the icon more cleverly, this is fine.